api/provider/aws/test_aws/service_create_test.go

```diff
@@ -0,0 +1,211 @@
+package test_aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestServiceCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "dpl_arn",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	defer provider.SetEntityIDGenerator("svc_id")()
+
+	loadBalancerInput := &elb.DescribeLoadBalancersInput{}
+	loadBalancerInput.SetLoadBalancerNames([]*string{aws.String("l0-test-lb_id")})
+	loadBalancerInput.SetPageSize(1)
+
+	listener := &elb.Listener{}
+	listener.SetInstancePort(80)
```

Comment:
My PR isn't merged yet, but I have a [helper function](https://github.com/quintilesims/layer0/pull/410/files#diff-de7695ca79b6fee67c84ed606a7efce9R70) and would also allow you to pass in a [default port](https://github.com/quintilesims/layer0/pull/410/files#diff-e30be2de824c2c73ed65e40c0652fb89R128) and would allow you to call the `SetListener` method instead.

In case it's awkward to simply set a hardcoded instance port.

Reply:
@jparsons04 That's pretty cool - my only hesitation is that all I need is an `InstancePort` set, not a full port. Do you think there's greater value in using a fully-formed default port, or in keeping as narrow of a test as possible by using bare-minimum objects?

Reply:
If you need only to set the instance port, then you could just use the default's ContainerPort: `provider.DefaultLoadBalancerPort.ContainerPort`

Reply:
Sorry, the HostPort, not ContainerPort

Reply:
✅